sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -129,6 +138,23 @@ int gnrc_lorawan_parse_dl(gnrc_lorawan_t *mac, uint8_t *buf, size_t len,
         }
     }
 
+    uint32_t _fcnt =
+        gnrc_lorawan_fcnt_stol(mac->mcps.fcnt_down, _hdr->fcnt.u16);
+
+    uint32_t fcnt_down = mac->mcps.fcnt_down;
+#if IS_USED(MODULE_GNRC_LORAWAN_1_1)
+    if (pkt->port) {
+        fcnt_down = mac->mcps.afcnt_down;
+    }
+#endif
```

Comment:
We should really avoid these #if macros here. If you can create a function `gnrc_lorawan_afcnt_down` and put there the #if macro it would simplify this snippet.